bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -231,15 +232,18 @@ public void changeStateTo(Item item, State state) {
                 event.valuecolor = itemUIRegistry.getValueColor(w);
                 event.widgetId = itemUIRegistry.getWidgetId(w);
                 event.visibility = itemUIRegistry.getVisiblity(w);
-                // event.item contains data from the item including its state (in event.item.state)
+                // event.item contains the (potentially changed) data of the item belonging to
+                // the widget including its state (in event.item.state)
+                final Item itemToBeSent = itemBelongsToWidget ? item : getItemForWidget(w);
```

Comment:
@maniac103 May I ask you to take a second look at these changes? The newly introduced method `getItemForWidget()` may return a `null` value. But later in the code we pass the variable `itemToBeSent` towards two methods line [242](https://github.com/openhab/openhab-core/pull/691/files#diff-78e437b3d2798d795dfa76249a91c583R242) and line [246](https://github.com/openhab/openhab-core/pull/691/files#diff-78e437b3d2798d795dfa76249a91c583R246)) which require a `@NonNull` value.

This could e.g. lead to a NPE like this:

```
2019-06-25 15:19:17.216 [WARN ] [pse.smarthome.core.items.GenericItem] - failed notifying listener 'org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener@71c0a5b7' about state update of item testString: null
java.lang.NullPointerException: null
	at org.eclipse.smarthome.core.items.dto.ItemDTOMapper.fillProperties(ItemDTOMapper.java:152) ~[137:org.openhab.core:2.5.0.201906250301]
	at org.eclipse.smarthome.core.items.dto.ItemDTOMapper.map(ItemDTOMapper.java:138) ~[137:org.openhab.core:2.5.0.201906250301]
	at org.eclipse.smarthome.io.rest.core.item.EnrichedItemDTOMapper.map(EnrichedItemDTOMapper.java:56) ~[?:?]
	at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.constructSitemapEvents(PageChangeListener.java:242) ~[?:?]
	at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.constructSitemapEvents(PageChangeListener.java:216) ~[?:?]
	at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.constructAndSendEvents(PageChangeListener.java:174) ~[?:?]
	at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.stateChanged(PageChangeListener.java:188) ~[?:?]
	at org.eclipse.smarthome.core.items.GenericItem$1.run(GenericItem.java:262) [137:org.openhab.core:2.5.0.201906250301]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
```

Reproducible by instructions in #897.

Reply:
@cweitkamp In that case we probably need to omit `event.item` and `event.state`, which is what I wanted to do to begin with, but concerns were raised on whether events without item are handled properly everywhere. I can't think of any other solution if the widget in question actually does not have any item though.
@lolodomo Thoughts?